source/Symbol/SwiftASTContext.cpp

```diff
@@ -4087,6 +4087,13 @@ SwiftASTContext::GetTypeFromMangledTypename(const char *mangled_typename,
                      .getPointer();
 
     if (found_type) {
+      // If we have an inout type at the top level, turn it into an lvalue type.
+      // Function parameters that are inout are treated the same as mutable vars
+      // here.
+      if (auto *inout_type = found_type->getAs<swift::InOutType>()) {
+        found_type = swift::LValueType::get(inout_type->getObjectType());
+      }
```

Comment:
Thanks for tackling this! Do you understand better why turning it into an LValue type is the right thing to do here? I'm asking because down in ...

Reply:
line SwiftASTContext::GetNumChildren() we have a really odd workaround for Classes that are LValue types and I wonder if what you want is that workaround.

Reply:
Hmm.. I don't know. I tried cloning that workaround for InOutTypes but that alone does not fix the test (but there many switch(TypeKind)s so I might have missed one. Doing it this way seems harmless and it's documented so let's do it.

Reply:
I don't claim to understand most of what goes on in SwiftASTContext, but it does have the general feeling that there are workarounds that only exist because of workarounds elsewhere. With LValueType, I think lldb should not be using it to represent anything at all. It's an implementation detail of the type checker and we would like to eliminate it from the type system altogether. What's going on here is that lldb is constructing LValueTypes from whole cloth and passing them around, checking for LValueTypes elsewhere within lldb. It should use some other mechanism for this purpose. At least now, TypeReconstruction and mangling never accept or produce LValueTypes, so the "badness" is entirely isolated inside lldb.